src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

```diff
@@ -0,0 +1,236 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Net.Test.Common;
+using System.Text;
+using System.Threading;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Sockets.Tests
+{
+    public class UnixDomainSocketTest
+    {
+        private readonly ITestOutputHelper _log;
+
+        public UnixDomainSocketTest(ITestOutputHelper output)
+        {
+            _log = TestLogging.GetInstance();
+        }
+
+        private void OnConnectAsyncCompleted(object sender, SocketAsyncEventArgs args)
+        {
+            ManualResetEvent complete = (ManualResetEvent)args.UserToken;
+            complete.Set();
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void Socket_CreateUnixDomainSocket_Throws_OnWindows()
+        {
+            Assert.False(Capability.UnixDomainSocketsSupport());
+            Assert.Throws<SocketException>(() => new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified));
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        public void Socket_ConnectAsyncUnixDomainSocketEndPoint_Success()
+        {
+            Assert.True(Capability.UnixDomainSocketsSupport());
+
+            string path = null;
+            SocketTestServer server = null;
+            UnixDomainSocketEndPoint endPoint = null;
+
+            for (int attempt = 0; attempt < 5; attempt++)
+            {
+                path = GetRandomNonExistingFilePath();
+                endPoint = new UnixDomainSocketEndPoint(path);
+                try
+                {
+                    server = SocketTestServer.SocketTestServerFactory(endPoint, ProtocolType.Unspecified);
+                    break;
+                }
+                catch
```

Comment:
Catch the expected exception type please. Catching generic exceptions is a bad programming pattern for multiple reasons.